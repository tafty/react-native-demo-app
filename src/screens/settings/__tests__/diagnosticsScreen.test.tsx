import 'react-native'
import React from 'react'
import { render } from 'react-native-testing-library'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import DiagnosticsScreen from '../diagnosticsScreen'

const mockStore = configureStore()

jest.mock('../../../metadata.json', () => {
  return {
    git_commit: 'dd8ad0e',
    build_date: '2020-02-10T17:18:17.375+0000',
  }
})

describe('Diagnostics Screen', () => {
  it('Should render values from state', () => {
    const componentId = 'component1'
    const localApi = 'localhost://uri'
    const remoteApi = 'https://uri'
    const environment = 'test'
    const initialState = {
      configuration: {
        localConfig: {
          api: {
            remoteConfigUri: localApi,
            isStubbed: false,
            isRemoteConfigEnabled: true,
          },
          environment,
        },
        remoteConfig: {
          api: {
            uri: remoteApi,
          },
        },
      },
    }

    const store = mockStore(initialState)

    const { getByTestId, toJSON } = render(
      <Provider store={store}>
        <DiagnosticsScreen
          componentId={componentId}
          apiUri=""
          configApiUri=""
          environment=""
          isApiStubbed={false}
          isRemoteConfigEnabled={false}
          realmPath=""
        />
      </Provider>,
    )

    expect(getByTestId('Configuration API Location-value').props.children).toBe(
      localApi,
    )
    expect(getByTestId('Environment-value').props.children).toBe(environment)
    expect(getByTestId('API End Point-value').props.children).toBe(remoteApi)
    expect(getByTestId('Device Brand-value').props.children).toBe('Apple')
    expect(getByTestId('Device Model-value').props.children).toBe('iPhone 6')
    expect(getByTestId('App Version-value').props.children).toBe('1.0.2')
    expect(getByTestId('Build Number-value').props.children).toBe('10000002')
    expect(getByTestId('Device ID-value').props.children).toBe(
      'this_is_a_unique_device_id',
    )
    expect(getByTestId('Git Hash-value').props.children).toBe('dd8ad0e')
    expect(getByTestId('Build Date-value').props.children).toBe(
      'Mon Feb 10 2020 17:18:17 GMT+0000',
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('Should inform that stubs are in use', () => {
    const componentId = 'component1'
    const localApi = 'localhost://uri'
    const remoteApi = 'https://uri'
    const environment = 'test'
    const initialState = {
      configuration: {
        localConfig: {
          api: {
            remoteConfigUri: localApi,
            isStubbed: true,
            isRemoteConfigEnabled: false,
          },
          environment,
        },
        remoteConfig: {
          api: {
            uri: remoteApi,
          },
        },
      },
    }

    const store = mockStore(initialState)

    const { getByTestId, toJSON } = render(
      <Provider store={store}>
        <DiagnosticsScreen
          componentId={componentId}
          apiUri=""
          configApiUri=""
          environment=""
          isApiStubbed={false}
          isRemoteConfigEnabled={false}
          realmPath=""
        />
      </Provider>,
    )

    expect(getByTestId('Configuration API Location-value').props.children).toBe(
      'Using internal stub',
    )
    expect(getByTestId('Environment-value').props.children).toBe(environment)
    expect(getByTestId('API End Point-value').props.children).toBe(
      'Using internal stub',
    )
    expect(getByTestId('Device Brand-value').props.children).toBe('Apple')
    expect(getByTestId('Device Model-value').props.children).toBe('iPhone 6')
    expect(getByTestId('App Version-value').props.children).toBe('1.0.2')
    expect(getByTestId('Build Number-value').props.children).toBe('10000002')
    expect(getByTestId('Device ID-value').props.children).toBe(
      'this_is_a_unique_device_id',
    )
    expect(getByTestId('Git Hash-value').props.children).toBe('dd8ad0e')
    expect(getByTestId('Build Date-value').props.children).toBe(
      'Mon Feb 10 2020 17:18:17 GMT+0000',
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
