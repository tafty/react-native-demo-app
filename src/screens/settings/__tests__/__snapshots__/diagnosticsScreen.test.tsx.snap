// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagnostics Screen Should inform that stubs are in use 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "rgba(246, 246, 246, 1)",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Configuration API Location-title"
      >
        Configuration API Location
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Configuration API Location-value"
      >
        Using internal stub
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Environment-title"
      >
        Environment
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Environment-value"
      >
        test
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="API End Point-title"
      >
        API End Point
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="API End Point-value"
      >
        Using internal stub
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device Brand-title"
      >
        Device Brand
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device Brand-value"
      >
        Apple
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device Model-title"
      >
        Device Model
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device Model-value"
      >
        iPhone 6
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device OS-title"
      >
        Device OS
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device OS-value"
      >
        iOS
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device OS Version-title"
      >
        Device OS Version
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device OS Version-value"
      >
        12.0
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="App Version-title"
      >
        App Version
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="App Version-value"
      >
        1.0.2
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Build Number-title"
      >
        Build Number
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Build Number-value"
      >
        10000002
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device ID-title"
      >
        Device ID
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device ID-value"
      >
        this_is_a_unique_device_id
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Git Hash-title"
      >
        Git Hash
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Git Hash-value"
      >
        dd8ad0e
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Build Date-title"
      >
        Build Date
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Build Date-value"
      >
        Mon Feb 10 2020 17:18:17 GMT+0000
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Diagnostics Screen Should render values from state 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "rgba(246, 246, 246, 1)",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Configuration API Location-title"
      >
        Configuration API Location
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Configuration API Location-value"
      >
        localhost://uri
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Environment-title"
      >
        Environment
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Environment-value"
      >
        test
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="API End Point-title"
      >
        API End Point
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="API End Point-value"
      >
        https://uri
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device Brand-title"
      >
        Device Brand
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device Brand-value"
      >
        Apple
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device Model-title"
      >
        Device Model
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device Model-value"
      >
        iPhone 6
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device OS-title"
      >
        Device OS
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device OS-value"
      >
        iOS
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device OS Version-title"
      >
        Device OS Version
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device OS Version-value"
      >
        12.0
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="App Version-title"
      >
        App Version
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="App Version-value"
      >
        1.0.2
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Build Number-title"
      >
        Build Number
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Build Number-value"
      >
        10000002
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Device ID-title"
      >
        Device ID
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Device ID-value"
      >
        this_is_a_unique_device_id
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Git Hash-title"
      >
        Git Hash
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Git Hash-value"
      >
        dd8ad0e
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(246, 246, 246, 1)",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
        testID="Build Date-title"
      >
        Build Date
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
        testID="Build Date-value"
      >
        Mon Feb 10 2020 17:18:17 GMT+0000
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
